lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -721,7 +721,7 @@ public function load(array $criteria, $entity = null, $assoc = null, array $hint
         $hydrator = $this->em->newHydrator($this->currentPersisterContext->selectJoinSql ? Query::HYDRATE_OBJECT : Query::HYDRATE_SIMPLEOBJECT);
         $entities = $hydrator->hydrateAll($stmt, $this->currentPersisterContext->rsm, $hints);
 
-        return $entities ? $entities[0] : null;
+        return $entities && isset($entities[0]) ? $entities[0] : null;
```

Comment:
In this case I think checking if entities == true, would be redundant because it will always return true if the isset returns true.